metric_providers/psu/energy/ac/powerspy2/machine/provider.py

```diff
@@ -13,3 +13,16 @@ def __init__(self, resolution, skip_check=False):
             skip_check=skip_check,
         )
         self._extra_switches = ['-u','mJ']
+
+    def check_system(self):
+        file_path = "/dev/rfcomm0"
+        if os.path.exists(file_path):
+            try:
+                with open(file_path, 'r', encoding='utf-8') as file:
+                    file.read()
+            except PermissionError as exc:
+                raise MetricProviderConfigurationError(f"{self._metric_name} provider could not be started.\nCannot read device at {file_path}.\n\nAre you running in a VM / cloud / shared hosting?\nIf so please disable the {self._metric_name} provider in the config.yml") from exc
+        else:
+            raise MetricProviderConfigurationError(f"{self._metric_name} provider could not be started.\nCould not find device at {file_path}.\n\nAre you running in a VM / cloud / shared hosting? \nIf so please disable the {self._metric_name} provider in the config.yml")
+
+        self.check_parallel_provider_running()
```

Comment:
why not the base function?

Reply:
Because I'm overwriting the check_system with a separate check - (checking the /dev/rfcomm0 filepath)

I'm not sure here if I'm missing something obvious, or if this file was just confusing the read fully on the github pr?